codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -215,9 +216,7 @@ public int onDataRead(final ChannelHandlerContext ctx, int streamId, ByteBuf dat
             // Check if we received a data frame for a stream which is half-closed
             Http2Stream stream = connection.requireStream(streamId);
 
-            verifyEndOfStreamNotReceived(stream);
             verifyGoAwayNotReceived();
-            verifyRstStreamNotReceived(stream);
 
             // We should ignore this frame if RST_STREAM was sent or if GO_AWAY was sent with a
             // lower stream ID.
```

Comment:
Yes we still want to count toward flow control for most instances.  Here are some relevant snippets from the spec:

[section 5.1](https://tools.ietf.org/html/draft-ietf-httpbis-http2-16#section-5.1):
>closed:
...
> Flow controlled frames (i.e., DATA) received after sending
      RST_STREAM are counted toward the connection flow control window.
      Even though these frames might be ignored, because they are sent
      before the sender receives the RST_STREAM, the sender will
      consider the frames to count against the flow control window.

[sectoin 5.4.2](https://tools.ietf.org/html/draft-ietf-httpbis-http2-16#section-5.4.2)
> A RST_STREAM is the last frame that an endpoint can send on a stream.
   The peer that sends the RST_STREAM frame MUST be prepared to receive
   any frames that were sent or enqueued for sending by the remote peer.
   These frames can be ignored, except where they modify connection
   state (such as the state maintained for header compression
   (Section 4.3), or flow control).

[section 6.8](https://tools.ietf.org/html/draft-ietf-httpbis-http2-16#section-6.8)
> After sending a GOAWAY frame, the sender can discard frames for
   streams with identifiers higher than the identified last stream.
   However, any frames that alter connection state cannot be completely
   ignored.  For instance, HEADERS, PUSH_PROMISE and CONTINUATION frames
   MUST be minimally processed to ensure the state maintained for header
   compression is consistent (see Section 4.3); similarly DATA frames
   MUST be counted toward the connection flow control window.  Failure
   to process these frames can cause flow control or header compression
   state to become unsynchronized.